packages/@react-stately/selection/src/SelectionManager.ts

```diff
@@ -274,14 +277,16 @@ export class SelectionManager implements MultipleSelectionManager {
     let keys: Key[] = [];
     let addKeys = (key: Key) => {
       while (key) {
-        let item = this.collection.getItem(key);
-        if (item.type === 'item') {
-          keys.push(key);
-        }
-
-        // Add child keys. If cell selection is allowed, then include item children too.
-        if (item.hasChildNodes && (this.allowsCellSelection || item.type !== 'item')) {
-          addKeys([...item.childNodes][0].key);
+        if (!this.state.disabledKeys.has(key)) {
+          let item = this.collection.getItem(key);
+          if (item.type === 'item') {
+            keys.push(key);
+          }
+
+          // Add child keys. If cell selection is allowed, then include item children too.
+          if (item.hasChildNodes && (this.allowsCellSelection || item.type !== 'item')) {
+            addKeys([...item.childNodes][0].key);
+          }
```

Comment:
Looked into an issue @snowystinger brought to my attention where hitting the escape key in a Table that supports selection always causes `onSelectionChange` to fire, even if the selection state was already empty. Looks like this happens because of https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/selection/src/SelectionManager.ts#L307-L309, which sends a new empty `Selection` set to `useControlledState`. 

I can go ahead and fix this by adding some form of deep set equality check to `useControlledState` but just wanna to make sure this is the desired fix/the current behavior isn't actually desired